cmd/manager/main.go

```diff
@@ -264,3 +272,101 @@ func serveCRMetrics(cfg *rest.Config) error {
 	}
 	return nil
 }
+
+func giveUsersPermissionToInteractWithCRs(cfg *rest.Config, ns string) error {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mobile-developer",
+			Namespace: ns,
+		},
+		Rules: rules(),
+	}
+
+	rolebinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mobile-developer",
+			Namespace: ns,
+		},
+		RoleRef:  roleRef(),
+		Subjects: subjects(),
+	}
+
+	kclient, err := client.New(cfg, client.Options{})
+	if err != nil {
+		return err
+	}
+
+	op, err := controllerutil.CreateOrUpdate(context.TODO(), kclient, role, func(ignore kruntime.Object) error {
+		role.Rules = rules()
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	log.Info("Role reconciled:", "Role", role.ObjectMeta.Name, "Operation", op)
```

Comment:
I may open a PR to the operator-sdk to start a discussion about using this `CreateOrUpdate` function in their examples and scaffolded controllers, since I think it's a better way than they currently do (for secondary resources at least). I think it's something that would enable us to clean up our reconcile functions a bit, and such a discussion upstream could inform that.